e2e/stack_test.go

```diff
@@ -99,6 +104,58 @@ func TestE2E(t *testing.T) {
 
 	const targetHeight = 5
 
+	// Hang until L1 responsive.
+	require.True(t, stackConfig.L1URL.IsReachable(ctx))
+
+	l1RPCClient, err := rpc.DialContext(ctx, stackConfig.L1URL.String())
+	require.NoError(t, err)
+	l1Client := e2e.NewL1Client(l1RPCClient)
+
+	// instantiate L1 user, tx signer.
+	user := stackConfig.Users[0]
+	signer := types.NewEIP155Signer(stackConfig.L1ChainID)
+
+	// op Portal
+	portal, err := bindings.NewOptimismPortal(stackConfig.DepositContractAddress, l1Client)
+	require.NoError(t, err)
+
+	// send user Deposit Tx
+	nonce, err := l1Client.Client.NonceAt(ctx, user.Address, nil)
+	require.NoError(t, err)
+
+	price, err := l1Client.Client.SuggestGasPrice(context.Background())
+	require.NoError(t, err)
+
+	gasPrice := new(big.Int).Mul(price, big.NewInt(2))
+	if gasPrice.BitLen() > 256 {
+		gasPrice = price // fallback to original price if overflow
+	}
+
+	depositTx, err := portal.DepositTransaction(
+		&bind.TransactOpts{
+			From: user.Address,
+			Signer: func(addr common.Address, tx *types.Transaction) (*types.Transaction, error) {
+				signed, err := types.SignTx(tx, signer, user.PrivateKey)
+				if err != nil {
+					return nil, err
+				}
+				return signed, nil
+			},
+			Nonce:    big.NewInt(int64(nonce)),
+			GasPrice: big.NewInt(price.Int64() * 2),
+			GasLimit: 1e6,
```

Comment:
Why is `1e6` used here for gas limit? I see that the gas limit is also passed in as `stackConfig.Genesis.SystemConfig.GasLimit/10` down below

Reply:
Mostly trial and error. The `1e6` here is the gasLimit for the L1 tx. It has since changed to `1e7`, which fixes a revert.

The other gasLimit is for L2 processing of the deposit transaction.

Reply:
Ah ok thanks. Could you pull it out into a constant for readability?

Reply:
Refactored in 71d7221, but there's the interplay between L1 and L2 gasLimit values that discourages me from pulling either as a constant.

(L1 gas gets burned according to the l2Gaslimit as part of the cross-chain gas accounting.)